src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
+        // Try to immediately build element, it may already be ready.
+        resource.build();
+        this.schedulePass();
+      } else {
+        this.pendingBuildResources_.push(element);
+        this.buildReadyResources_();
+      }
     }
 
     dev.fine(TAG_, 'element added:', resource.debugid);
   }
 
   /**
+   * Builds resources that are ready to be built.
+   * @private
+   */
+  buildReadyResources_() {
+    const builtResourcesIndexes = [];
+    this.pendingBuildResources_.forEach((resource, index) => {
+      if (this.documentReady_ || isElementFullyParsed(resource.element)) {
+        resource.build();
+        builtResourcesIndexes.push(index);
+      }
+    });
+
+    // Remove resources that we just built from pending resources.
+    for (let i = builtResourcesIndexes.length; i > 0; i--) {
```

Comment:
`builtResourcesIndexes.length - 1`, or else `i` will be out of bounds on the first run.